packages/mui-material/src/Checkbox/Checkbox.js

```diff
@@ -41,15 +41,17 @@ const CheckboxRoot = styled(SwitchBase, {
     ];
   },
 })(({ theme, ownerState }) => ({
-  color: theme.palette.text.secondary,
+  color: (theme.vars || theme).palette.text.secondary,
   ...(!ownerState.disableRipple && {
     '&:hover': {
-      backgroundColor: alpha(
-        ownerState.color === 'default'
-          ? theme.palette.action.active
-          : theme.palette[ownerState.color].main,
-        theme.palette.action.hoverOpacity,
-      ),
+      backgroundColor: theme.vars
+        ? `rgba(${theme.vars.palette.text.primaryChannel} / ${theme.vars.palette.action.hoverOpacity})`
```

Comment:
You need to add here too the check for the `ownerState.color` that you have below.

Reply:
We don't have `action.activeChannel` yet, seems like we have to add it or find alternatives.

Reply:
I have fixed the hover background color by applying using `palette.primary.mainChannel` and `palette.action.hoverOpacity`.

How do I do the check for `ownerState.color`, I didn't understand that part ?

Reply:
It's the same as in https://github.com/mui/material-ui/pull/32599#discussion_r864769251, let's wait for it to be resolved there and we can fix it in this PR too.

Reply:
Note, I just merged https://github.com/mui/material-ui/pull/32599. Feel free to rebase and you will have the `activeChannel` in your theme.